lbrynet/reflector/server/server.py

```diff
@@ -67,71 +133,169 @@ def _get_valid_response(self, response_msg):
 
     def handle_request(self, request_dict):
         if self.received_handshake is False:
-            return self.handle_handshake(request_dict)
-        else:
-            return self.handle_normal_request(request_dict)
-
-    def handle_handshake(self, request_dict):
-        log.debug('Handling handshake')
-        if 'version' not in request_dict:
-            raise ValueError("Client should send version")
-        self.peer_version = int(request_dict['version'])
-        if self.peer_version != 0:
-            raise ValueError("I don't know that version!")
+            if VERSION not in request_dict:
+                raise ReflectorRequestError("Client should send version")
+            peer_version = int(request_dict[VERSION])
+            return self.handle_handshake(peer_version)
+
+        if SD_BLOB_HASH in request_dict and SD_BLOB_SIZE in request_dict:
```

Comment:
change this to read:

```
if self.is_descriptor_request(request_dict):
    return self.handle_descriptor_request(request_dict)
if self.is_blob_request(request_dict):
    return self.handle_blob_request(request_dict)
```

where `is_descriptor_request` is
```
def is_descriptor_request(request_dict):
    return SD_BLOB_HASH in request_dict and SD_BLOB_SIZE in request_dict
```

etc...
return